### PR TITLE
Added StreamUpload

### PR DESCRIPTION
**Pull request by cnlangzi**
link to 0chain/blobber#166

add parameters on ./zbox upload

- stream: true/false.  false is default
- chunksize: int.  64KB is default

### CODE REVIEW

**Comment by cnlangzi** [comment]
it works with partial stream because of `Avoids outsourcing attach` with merkle tree now.
- update with stream mode
- hash is still ask data is fully loaded in memory for hashing

I would suggest to review PR, and get it passed first. Because there are too much changes now.  Let us follow it on https://github.com/0chain/zboxcli/issues/36

**Comment by cnlangzi** [comment]
@Sriep please help me to review PR if you get some time left , thanks.

File: cmd/root.go [hunk 1]
```diff
@@ -28,6 +29,7 @@ var walletClientKey string
 var cDir string
 var bSilent bool
 var allocUnderRepair bool
+var devserver bool
```
Comment (Sriep) [inline cmd/root.go:32]: Camel case `devServer`.

File: cmd/upload.go [hunk 2]
```diff
@@ -66,19 +69,72 @@ var uploadCmd = &cobra.Command{
...
-				err = allocationObj.EncryptAndUploadFileWithThumbnail(localpath, remotepath, thumbnailpath, attrs, statusBar)
-			} else {
-				err = allocationObj.UploadFileWithThumbnail(localpath, remotepath, thumbnailpath, attrs, statusBar)
+		if stream {
+
+			chunkSize, _ := cmd.Flags().GetInt("chunksize")
+
+			fileReader, err := os.Open(localpath)
+			if err != nil {
+				goto ThrownErr
```
Comment (Sriep) [inline cmd/upload.go:78]: Please don't use `goto. Find another way. 
For instance you could add the `ThrowErr`code to a `defer`  block and replace the  `goto ThrowErr`  with  `return`.

**Review by Sriep — CHANGES_REQUESTED** [review]

**Comment by cnlangzi** [comment]
ignore it first. thanks.

**Pull request closed without merging**